test/evergreen.yml

```diff
@@ -307,11 +310,17 @@ functions:
           powershell.exe  -NonInteractive '.\test\evergreen\build_windows.ps1' -configure 1 $DEFINED_EVERGREEN_CONFIG_FLAGS ${windows_configure_flags|}
         else
           echo "Using config flags $DEFINED_EVERGREEN_CONFIG_FLAGS ${posix_configure_flags|}"
-          # Fetch the gperftools library if needed. This will also get tcmalloc.
+
           if [[ ${ENABLE_TCMALLOC|0} -eq 1 ]]; then
-            is_cmake_build=true
-            . ../automation-scripts/evergreen/find_gperftools.sh ${s3_access_key} ${s3_secret_key} ${build_variant} $is_cmake_build
+            # Run this script in its own process as it depends on managing
```

Comment:
Suggested safety check
```
if [[ ${CMAKE_BUILD_TYPE|} = "ASan" || ${CMAKE_BUILD_TYPE|} = "MSan" || ${CMAKE_BUILD_TYPE|} = "TSan" ]]; then
    # FIXME-XXXX - Remove this once we have sanitizer-compatible TCMalloc builds
    # UBSan works with TCMalloc and is not excluded
    echo "TCMalloc is not currently supported when running with sanitizers"
    exit 1
fi
```

Reply:
Can we use a regex with `(A|M|T|U)San$`?

Reply:
Yeah that looks better. `~=` is a bash-ism, but we've explicitly set `shell: bash` above so no worries on that front.
I double checked this and I've messed up the comparison above. I think it should instead be
```
san_regex="-DCMAKE_BUILD_TYPE=(A|M|T|UB)San$"
if [[ "${CMAKE_BUILD_TYPE|}" =~ $san_regex ]]; then
```